businessRules/donations.js

```diff
@@ -0,0 +1,47 @@
+const { v4 } = require('uuid');
+const { createDonation } = require('../models/donations');
+const { updateIdea, getAllIdeas } = require('../models/ideas');
+
+function createDonationsWithDataValidation(firstName, sum, userId, ideaId) {
+    const generateId = v4();
+    const anonymous = 'Anonymous';
+
+    if (!firstName && !userId) {
+        firstName = anonymous;
+        userId = generateId;
+    }
+
+    if (!userId) {
+        userId = generateId;
+    }
+
+    if (!firstName) {
+        firstName = anonymous;
+    }
```

Comment:
You don't need 3 `if`s here - try to refactor it to 2 `if`s. You can then simplify further - and remove `if`s altogether.

Reply:
I've been thinking, and replaced it with
firstName = firstName || Anonymous

Updated and tested.